event/src/main/java/com/bazaarvoice/emodb/event/db/astyanax/AstyanaxEventWriterDAO.java

```diff
@@ -78,16 +89,23 @@ public Void apply(BatchUpdate.Row<ByteBuffer, Integer> row) {
                                 }
                             });
 
-                    for (int i = 0; i < allocation.getLength(); i++) {
-                        int eventIdx = allocation.getOffset() + i;
+                    int eventCount = 0;
+                    while (eventsIter.hasNext()) {
                         ByteBuffer eventData = eventsIter.next();
+                        eventCount += 1;
+                        int eventIdx = allocation.getOffset() + eventCount;
                         row.putColumn(eventIdx, eventData, Ttls.toSeconds(eventTtl, 1, null));
 
                         if (sink != null && !sink.accept(AstyanaxEventId.create(channel, slabId, eventIdx), eventData)) {
                             sink = null;  // Sink isn't interested in more events.
                         }
                     }
 
+                    // the iterator should have as many entries as allocation, if not an error has occurred
+                    if (eventCount != allocation.getLength()) {
+                        throw new RuntimeException("AstyanaxEventWriterDAO.addAll: allocation size (" + allocation.getLength() + ") not equal to number of events in iterator (" + eventCount + ")");
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Define and throw a dedicated exception instead of using a generic one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.qa.bazaarvoice.com/coding_rules#rule_key=squid%3AS00112)